scripts/img_check.sh

```diff
@@ -351,27 +351,19 @@ function checkFirewall {
         ((WARN++))
       fi
     elif [[ $OS == "CentOS Linux" ]]; then
-      if [ -f /usr/lib/systemd/system/csf.service ]; then
+      if [ -f /etc/csf/csf.conf ]; then
         fw="csf"
-        if [[ $(systemctl status $fw >/dev/null 2>&1) ]]; then
-          
-        FW_VER="\e[32m[PASS]\e[0m Firewall service (${fw}) is active\n"
-        ((PASS++))
-        elif cmdExists "firewall-cmd"; then
-          if [[ $(systemctl is-active firewalld >/dev/null 2>&1 && echo 1 || echo 0) ]]; then
-           FW_VER="\e[32m[PASS]\e[0m Firewall service (${fw}) is active\n"
+        csfa=$(csf -l |head -1 |sed "s/ .*//")
+        if [[ $csfa == "iptables" ]]; then
+          FW_VER="\e[32m[PASS]\e[0m Firewall service (${fw}) is active\n"
           ((PASS++))
-          else
-            FW_VER="\e[93m[WARN]\e[0m No firewall is configured. Ensure ${fw} is installed and configured\n"
-          ((WARN++))
-          fi
         else
           FW_VER="\e[93m[WARN]\e[0m No firewall is configured. Ensure ${fw} is installed and configured\n"
         ((WARN++))
         fi
       else
         fw="firewalld"
-        if [[ $(systemctl is-active firewalld >/dev/null 2>&1 && echo 1 || echo 0) ]]; then
+        if [[ $(systemctl status firewalld >/dev/null 2>&1; echo $?) -eq '0' ]]; then
           FW_VER="\e[32m[PASS]\e[0m Firewall service (${fw}) is active\n"
```

Comment:
The following else states no firewall is installed, but in this logic step we could be in a state where firewalld is installed by not running. We should probably add a final catch that says no firewall is installed if there is no `csf` or `firewalld` detected.